torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -431,38 +447,84 @@ class _OnnxSchemaChecker:
 
     It provides methods to check for input compatibility based on the OpSchema. It also
     provides a matching score to indicate how well the OpSchema matches the input and
-    kwargs types.
+    kwargs types. A function will be evaluated as perfect match, nearest match eligible,
+    or no match.
 
-    There are three types of ONNX overloads in torchlib:
+    Here are some common examples in categories:
 
-    1. Different types: Caused by the difference between the ONNX spec and PyTorch.The
-        matching system finds the correct one.
+    1. [NOTE: Perfect match]: The number of inputs and attributes are exactly the same as
+        the OpSchema. The types of inputs and attributes are exactly the same as the
+        OpSchema.
 
         ```python
-        @torch_op("aten::mul")
-        def aten_mul(self: TReal, other: TReal) -> TReal:
+        inputs = (Tensor[2, 3], Tensor[2, 3])
+        attributes = {"alpha": 1.0}
+
+        @torch_op("aten::op")
+        def aten_op(self: TReal, other: TReal, alpha: float = 1) -> TReal:
             ...
 
-        @torch_op("aten::mul")
-        def aten_mul_bool(self: BOOL, other: BOOL) -> BOOL:
+        ```
+        Result: Perfect match.
+
+    2. [NOTE: Optional input]: The dispatcher recognizes optional inputs. However,
+        the input can't be ignored. None must be provided.
```

Comment:
nit: to check my understanding. Is this due to onnxscript limitation that None cannot be set as default value for `Optional[xxx]` typed argument? Such that `inputs = (Tensor([2, 3], )` will not be matched, but otherwise it should? If there is an issue to link here that will be great ...

I typed above but then I found convolution aten signature is `- func: convolution(Tensor input, Tensor weight, Tensor? bias, ...`. So the optional input by design does not have default value. And in that case, the type is only meaning `None | INT64` so it is syntactically incorrect to omit it.

Reply:
I think PyTorch doesn't omit any inputs, because if they do, we should see a lot of breakings now in terms of the `param_schema` separating. Not sure if there will be a change on this, but currently, we don't fill in default value to input, nor omit None.